project_event/models/project_task.py

```diff
@@ -123,6 +123,18 @@ def create(self, vals):
             elif vals['activity_task_type'] == 'task':
                 vals['code'] = self.env['ir.sequence'] \
                     .next_by_code('project.task.task')
+        return_create = super(Task, self).create(vals)
+        if vals['activity_task_type'] == 'activity':
+            vals['parent_id'] = return_create.id
+            vals['message_follower_ids'] = None
+            self.create_main_task(vals)
+        return return_create
+
+    @api.multi
+    def create_main_task(self, vals):
+        vals['activity_task_type'] = 'task'
+        vals['code'] = self.env['ir.sequence'] \
+            .next_by_code('project.task.task')
```

Comment:
@eilst : Code  looks redundant. Hope we can still optimize methods
e.g if vals['activity_task_type'] == 'activity':  it simply returns   self.create_main_task(vals) 
 if vals['activity_task_type'] == 'task'::  it simply returns  return_create on execution of def create(self, vals):

Reply:
done